sktime/datatypes/_hierarchical/_check.py

```diff
@@ -85,9 +85,20 @@ def check_pdmultiindex_hierarchical(obj, return_metadata=False, var_name="obj"):
         msg = f"{var_name} must have a MultiIndex, found {type(obj.index)}"
         return _ret(False, msg, None, return_metadata)
 
-    # check that columns are unique
+    # check that row index is unique
+    msg = f"{var_name} must have unique row indices, but found {obj.index}"
+    if not obj.index.is_unique:
+        return _ret(False, msg, None, return_metadata)
+
+    # check that rows are sorted
+    msg = f"{var_name} must have lexsorted row index, but found {obj.index}"
```

Comment:
Agree with @patrickzib, but if this will be refactored in #3512 then we can leave the copied code.

Reply:
yes, I'd leave it to then